### DISCUSSION TITLE
Small Projects : provide a better help system

### DISCUSSION THREAD

**Discussion by opcoach** [eclipse-ide/contributing#4] [Ideas]
The Eclipse IDE, in all of its provided packages is shipped with all the help plugins. This is cool because we can get the help automatically inside eclipse, but it costs a lot in term of download size. 

Actually do we access often to the inline Eclipse help ?  A public web version could be enough for the majority of users ? 

Let's run this command in a standard Eclipse Modeling install  (2024-09), considering that help and doc plugins are concerned : 

_du -ch *help* *doc*                                      
756K	org.eclipse.epp.mpc.help.ui_1.11.0.v20240709-0639.jar
312K	org.eclipse.help.base.source_4.4.500.v20240903-0240.jar
444K	org.eclipse.help.base_4.4.500.v20240903-0240.jar
184K	org.eclipse.help.source_3.10.400.v20240415-0528.jar
228K	org.eclipse.help.ui.source_4.7.100.v20240621-0541.jar
544K	org.eclipse.help.ui_4.7.100.v20240621-0541.jar
208K	org.eclipse.help.webapp.source_3.11.500.v20240606-0803.jar
752K	org.eclipse.help.webapp_3.11.500.v20240606-0803.jar
280K	org.eclipse.help_3.10.400.v20240415-0528.jar
1,2M	org.eclipse.mylyn.help.ui_4.4.0.v20240617-1534.jar
1,3M	org.eclipse.mylyn.wikitext.help.ui_4.4.0.v20240820-2036.jar
5,1M	org.eclipse.egit.doc_7.0.0.202409031743-r.jar
 15M	org.eclipse.emf.cdo.doc_4.2.12.v20240904-1115.jar
 12M	org.eclipse.emf.compare.doc_3.3.24.202401051648.jar
3,1M	org.eclipse.emf.doc_2.31.0.v20230828-0744.jar
2,0M	org.eclipse.emf.ecoretools.doc_3.5.1.202404261351.jar
 12K	org.eclipse.emf.parsley.doc.source_1.18.0.v20240826-1656.jar
860K	org.eclipse.emf.parsley.doc_1.18.0.v20240826-1656.jar
 20K	org.eclipse.emf.transaction.doc_1.5.0.202408231629.jar
 20K	org.eclipse.emf.validation.doc_1.5.0.202408231629.jar
 20K	org.eclipse.emf.workspace.doc_1.4.0.202408231629.jar
1,3M	org.eclipse.gef.cloudio.doc_5.0.2.202311221639.jar
5,9M	org.eclipse.gef.common.doc_5.0.3.202311221639.jar
1,0M	org.eclipse.gef.dot.doc_5.1.4.202311221639.jar
708K	org.eclipse.gef.fx.doc_5.0.5.202311221639.jar
3,6M	org.eclipse.gef.geometry.doc_5.0.3.202311221639.jar
1,1M	org.eclipse.gef.graph.doc_5.1.0.202311221639.jar
3,1M	org.eclipse.gef.layout.doc_5.0.0.202311221639.jar
 11M	org.eclipse.gef.mvc.doc_5.2.5.202311221639.jar
4,9M	org.eclipse.gef.zest.doc_5.1.4.202311221639.jar
7,7M	org.eclipse.jdt.doc.isv_3.14.2500.v20240827-0447.jar
7,2M	org.eclipse.jdt.doc.user_3.15.2300.v20240823-1014.jar
240K	org.eclipse.mylyn.github.doc_6.5.0.v20240617-1534.jar
 16K	org.eclipse.mylyn.wikitext.asciidoc.ui_4.4.0.v20240617-1534.jar
112K	org.eclipse.mylyn.wikitext.asciidoc_4.4.0.v20240630-0904.jar
4,3M	org.eclipse.ocl.doc_3.22.0.v20240902-1518.jar
320K	org.eclipse.oomph.p2.doc_1.14.0.v20230625-0755.jar
2,5M	org.eclipse.oomph.setup.doc_1.14.0.v20230618-0610.jar
6,0M	org.eclipse.pde.doc.user_3.16.0.v20240821-1908.jar
 41M	org.eclipse.platform.doc.isv_4.33.0.v20240827-0447.jar
4,0M	org.eclipse.platform.doc.user_4.33.0.v20240823-1014.jar
 16K	org.eclipse.uml2.doc_5.5.0.v20221116-1811.jar
1,7M	org.eclipse.xsd.doc_2.28.0.v20230614-0743.jar
152M	total_

So in this release, help and doc plugins occupies **152 Mb on 696 Mb (around 20 %)**  !! That's a lot...  Imagine that the millions of downloads save 150 Mb each time, we could really reduce the impact of downloads.

I have not made stats on other packages delivery, but it will be basically similar. 

On the other hand, the local help displayed in the browser is delivered by an internal server, running on the laptop and available on this URL : http://127.0.0.1:61015/help/index.jsp?token=833b21f2-a510-4e55-a845-619de90e7ad9 

Actually the millions of users will access to a local web site that could be hosted only once on the eclipse web site... 

And having all the help files locally is only needed by the org.eclipse.help.ui plugin (to provide inline help in the IDE). In this case, if the help is needed, a dialog could appear to propose to download it. 

The goal of this discussion is to define a small project that could be driven by students and to define the main tasks that could be done and the objectives to achieve. 

For instance : 

- identifying the amount of place wasted by the help in all the different packages
- finding differences between 'doc' and 'help' plugins
- creating an option to download the help plugins only if help is used using the help window instead of in browser
- changing the products definitions to remove the pure help plugins and add the new downloadable help
- generating the static help web site, for each version, to make it accessible on eclipseide.org

... 

This **small project**,  can contain all the tasks expected in a 'student project', such as : analysis, finding solutions, coding and testing and of course having discussion with the community to make it working. 

This discussion is opened to prepare it  in a more formal format.

**Comment by opcoach** [comment]
In addition to the existing static documentation we should consider to provide an internal RAG chatbot in Eclipse to help to deal with the help.

**Reply by danthe1st** [reply]
I want to note that many people have a big (and IMO justified) hate towards LLM chatbots (including RAG applications) included in many applications nowadays.
As an Open Source IDE, I think Eclipse should try to not annoy users with features like that and if these are built, they shouldn't be installed by default and there should not be any annoying buttons/banners/etc. for it.

**Reply by opcoach** [reply]
The idea is to provide another way to access the documentation.. asking a question in natural language instead of browsing tons of documentation ... My suggestion is not necessary to force these "many people" to use it because it would be provided by default.. a first step is to provide this kind of tooling on the market place and see how it is adopted...

**Reply by danthe1st** [reply]
I don't have anything against this approach as long as it isn't promoted (much) in the IDE outside the marketplace.
